### ISSUE TITLE
KeyExists error "deleted_txo_mmr_position_to_height_index"

### ISSUE THREAD

**Issue by sdbondi** [tari-project/tari#4833]
Occurred when submitting a mined block, multiple blocks were submitted quickly after each other

09:56 ERROR Failed to apply DB transaction: KeyExists { table_name: "deleted_txo_mmr_position_to_height_index", key: "7bd11b00" }
09:56 WARN  Failed to commit reorg chain: KeyExists { table_name: "deleted_txo_mmr_position_to_height_index", key: "7bd11b00" }. Restoring last chain.
09:56 ERROR Failed to apply DB transaction: KeyExists { table_name: "deleted_txo_mmr_position_to_height_index", key: "7bd11b00" }
09:56 WARN  Obscured status error: status: Internal, message: "Chain storage error: Key 7bd11b00 in deleted_txo_mmr_position_to_height_index already exists", details: [], metadata: MetadataMap { headers: {} }

**Comment by sdbondi** [comment]
[logs.zip](https://github.com/tari-project/tari/files/9849105/logs.zip)

**Comment by SWvheerden** [comment]
I think it might be related:
```
 // base_layer/core/src/chain_storage/async_db.rs:80 2022-10-24 08:06:28.853078000 [c::bn::comms_interface::inbound_handler] [cc1ade71a24700ddefc86457dbb6e1446229641b0e1c56bce909bcd0f1a0f950,7acd212bf3c3b5a0428ebfec90] WARN  Peer 8a970ecffa8b8ab1e2eb962b3b sent an invalid block: Validation error: Expected block previous hash to be 27992f11407698867c971e15d4751e539b32a3099a08ddbfd1358b23d7c801e5, but was 8a8d0b082a190ac1da274a6307ff86b3fb9b7b0d9ca4e598a161069e2d460b66
 // base_layer/core/src/base_node/comms_interface/inbound_handlers.rs:671 2022-10-24 08:06:28.853094000 [c::bn::base_node_service::service] [cc1ade71a24700ddefc86457dbb6e1446229641b0e1c56bce909bcd0f1a0f950,7acd212bf3c3b5a0428ebfec90] ERROR Failed to handle incoming block message: Comms interface error: `Chain storage error: Validation error: Expected block previous hash to be 27992f11407698867c971e15d4751e539b32a3099a08ddbfd1358b23d7c801e5, but was 8a8d0b082a190ac1da274a6307ff86b3fb9b7b0d9ca4e598a161069e2d460b66`
 // base_layer/core/src/base_node/service/service.rs:307 2022-10-24 08:06:28.853360000 [c::bn::chain_state_sync_service] [cc1ade71a24700ddefc86457dbb6e1446229641b0e1c56bce909bcd0f1a0f950,7acd212bf3c3b5a0428ebfec90] DEBUG Removing disconnected/banned peer `8a970ecffa8b8ab1e2eb962b3b` from chain metadata list 
 ```

**Comment by SWvheerden** [comment]
[base-node.zip](https://github.com/tari-project/tari/files/9858055/base-node.zip)
also happens in this log

**Comment by SWvheerden** [comment]
Might be fixed by: #4865

**Comment by SWvheerden** [comment]
could also be caused by #4866
